File: Data Structures & Algorithms/search-2d-matrix/submission-0.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int low = 0;
        int c = matrix[0].size();
        int r = matrix.size();
        int high = r*c - 1;
        while(low<=high){
            int mid = (high+low)/2;
            int temp = matrix[mid/c][mid%c];
            if(temp == target){
                return true;
            }
            else if(temp < target){
                low = mid + 1;
            }
            else{
                high = mid - 1;
            }
        }

        return false;
    }
};

File: Data Structures & Algorithms/permutation-string/submission-0.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        if(len(s1)>len(s2)):
            return False
        low = 0
        high = len(s1)-1
        cmp = Counter(s1)
        while(high<len(s2)):
            temp = s2[low:high+1]
            if(Counter(temp) == cmp):
                return True
            low = low + 1
            high = high + 1
        return False
        